### ISSUE TITLE
Initialize fittable transform parameters at transform initialization

### ISSUE THREAD

**Issue by alecgunny** [ML4GW/ml4gw#11]
For transforms like `RandomWaveformInjection` that have parameters that are expected to be fit on data (but not via gradient descent), initialize them with all 0s at object initialization time, that way they can load in weights from a checkpoint without having to get re-fit.

**Comment by alecgunny** [comment]
These parameters should also be registered as buffers, not as parameters. See [this discussion](https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723/11)

**Comment by alecgunny** [comment]
This also means that classes like `RandomWaveformInjection` will generally need to be _initialized_ with a `num_channels` or `num_ifos` argument rather than inferring it at `fit` time, which is probably a fair restriction.
